combined_robot_hw/CMakeLists.txt

```diff
@@ -1,33 +1,50 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(combined_robot_hw)
 
+# Load catkin and all dependencies required for this package
 find_package(catkin REQUIRED COMPONENTS
   hardware_interface
   pluginlib
   roscpp
 )
 
-include_directories(include ${catkin_INCLUDE_DIRS})
-
+# Declare a catkin package
 catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS hardware_interface pluginlib roscpp
+  INCLUDE_DIRS
+    include
+  LIBRARIES
+    ${PROJECT_NAME}
+  CATKIN_DEPENDS
+    hardware_interface
+    pluginlib
+    roscpp
 )
 
-add_library(${PROJECT_NAME}
-  src/combined_robot_hw.cpp
-)
+
+###########
+## Build ##
+###########
+
+# Specify header include paths
+include_directories(include ${catkin_INCLUDE_DIRS})
```

Comment:
what do you think about using `target_include_directories` instead of this global option?

Reply:
I am not a cmake & catkin expert, but my understanding is that within the framework of catkin, most of the benefits of `target_include_directories` are lost. Here is my understanding of the situation - Please correct me if I'm wrong:

The main "disadvantage" of `include_directories` is that it offers less granularity, affecting all targets in the current file. More importantly, it also affects all targets in files included via `add_subdirectory`. Since we don't have cmake subdirectories (thanks to catkin), we do not need to worry about this second point. The first point is valid, but the lack of control between targets doesn't seem to be a big deal IMO.

`target_include_directories` offers the ability to control what headers are exported to dependent targets. However, the `catkin_package(INCLUDE_DIRS)` here and `${catkin_INCLUDE_DIRS}` in dependent packages handles that for us.

In my (very non-expert) view, since all the benefits are already handled by catkin, `target_include_directories` would just clutter the `CMakeLists` without any benefits.

Reply:
Point taken. We are on the same page about this and indeed the even though the global approach is a little sloppier, it saves boilerplate that wouldn't change much anyways.